yb_migrate/cmd/exportSchema.go

```diff
@@ -39,49 +42,53 @@ to quickly create a Cobra application.`,
 	},
 
 	Run: func(cmd *cobra.Command, args []string) {
-		checkSchemaDirs()
 		exportSchema()
 	},
 }
 
 func exportSchema() {
-	utils.PrintIfTrue(fmt.Sprintf("export of schema for source type as '%s'\n", source.DBType), !source.GenerateReportMode)
-
+	if schemaIsExported(exportDir) {
+		if startClean {
+			proceed := utils.AskPrompt(
+				"CAUTION: Using --start-clean will overwrite any manual changes done to the " +
+					"exported schema. Do you want to proceed")
+			if !proceed {
+				return
+			}
+			for _, dirName := range []string{"schema", "reports", "temp", "metainfo/schema"} {
+				utils.CleanDir(filepath.Join(exportDir, dirName))
+			}
+			clearSchemaIsExported(exportDir)
+		} else {
+			fmt.Fprintf(os.Stderr, "Schema is already exported. "+
+				"Use --start-clean flag to export schema again -- "+
+				"CAUTION: Using --start-clean will overwrite any manual changes done to the exported schema.\n")
+			return
+		}
+	}
+	utils.PrintAndLog("export of schema for source type as '%s'\n", source.DBType)
 	// Check connection with source database.
 	err := source.DB().Connect()
 	if err != nil {
 		utils.ErrExit("Failed to connect to the source db: %s", err)
 	}
-
 	source.DB().CheckRequiredToolsAreInstalled()
-
-	if !source.GenerateReportMode {
-		migration.PrintSourceDBVersion(&source)
-	}
-
+	migration.PrintSourceDBVersion(&source)
 	migration.CreateMigrationProjectIfNotExists(&source, exportDir)
 
 	switch source.DBType {
 	case ORACLE:
-		utils.PrintIfTrue("preparing Ora2Pg for schema export from Oracle\n", source.VerboseMode, !source.GenerateReportMode)
-
 		migration.Ora2PgExtractSchema(&source, exportDir)
 	case POSTGRESQL:
-		utils.PrintIfTrue("preparing pg_dump for schema export from PG\n", source.VerboseMode, !source.GenerateReportMode)
-
 		migration.PgDumpExtractSchema(&source, exportDir)
 	case MYSQL:
-		utils.PrintIfTrue("preparing Ora2Pg for schema export from MySQL\n", source.VerboseMode, !source.GenerateReportMode)
-
 		migration.Ora2PgExtractSchema(&source, exportDir)
 	default:
-		fmt.Printf("Invalid source database type for export\n")
+		utils.ErrExit("Invalid source database type for export\n")
 	}
 
-	if !source.GenerateReportMode { //check is to avoid report generation twice via generateReport command
-		fmt.Printf("\nexported schema files created under directory: %s\n", exportDir+"/schema")
-		generateReport()
-	}
+	fmt.Printf("\nExported schema files created under directory: %s\n", exportDir+"/schema")
```

Comment:
nit: maybe now this can be converted to `utils.PrintAndLog()`

Reply:
Done.